### PR TITLE
Finished Game Router and factored out authentication into its own module

### PR DESCRIPTION
**Pull request by elimelt**
Title pretty much says it all. Basic REST endpoints for updating user's game data, plus some major clean up of how we handle authentication

### CODE REVIEW

File: server/controllers/games.js [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+    if (result.success) res.status(201).send({ success: true })
+    else
+      return res
+        .status(400)
+        .send({ error: 'Bad Request: Missing data in request body' })
+  } catch (error) {
+    console.error(error)
+    res.status(500).send('Internal Server Error')
+  }
+})
```
[the remark this answers is not in the document]
Comment (elimelt) [reply server/controllers/games.js:39]: For now don't worry about these. We can add rate limiting once we implement the front-end service for our game router.

File: server/controllers/games.js [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+    res.status(401).send({ success: false, error: 'Token unauthorized' })
+
+  try {
+    const result = await clearGameEntries(userID)
+    if (result.success) res.status(200).send(result)
+    else res.status(400).send({ error: 'could not get game entries' })
+  } catch (error) {
+    res.status(500).send({ error: 'Internal Server Error' })
+  }
+})
```
[the remark this answers is not in the document]
Comment (elimelt) [reply server/controllers/games.js:83]: see above

**Review by kaitinderr — APPROVED** [review]
The endpoints look good! We can address the rate limiting at another time.

**Review by Harshi-R — APPROVED** [review]
Everything looks great, thank you Elijah!

**Pull request merged**